src/analyze/analyze.c

```diff
@@ -1213,7 +1239,99 @@ static int dot(int argc, char *argv[], void *userdata) {
+                { "Color", "After",             config_parse_string,  0, &arg_dependency_color[UNIT_AFTER]     },
+                { "Color", "Requires",          config_parse_string,  0, &arg_dependency_color[UNIT_REQUIRES]  },
+                { "Color", "Requisite",         config_parse_string,  0, &arg_dependency_color[UNIT_REQUISITE] },
+                { "Color", "Wants",             config_parse_string,  0, &arg_dependency_color[UNIT_WANTS]     },
+                { "Color", "Conflicts",         config_parse_string,  0, &arg_dependency_color[UNIT_CONFLICTS] },
+                { "Color", "BindsTo",           config_parse_string,  0, &arg_dependency_color[UNIT_BINDS_TO]  },
+                { "Color", "PartOf",            config_parse_string,  0, &arg_dependency_color[UNIT_PART_OF]   },
+                { "Color", "OnFailure",         config_parse_string,  0, &arg_dependency_color[UNIT_ON_FAILURE]},
+                { "Color", "Triggers",          config_parse_string,  0, &arg_dependency_color[UNIT_TRIGGERS]  },
+                { "Style", "After",             config_parse_string,  0, &arg_dependency_style[UNIT_AFTER]     },
+                { "Style", "Requires",          config_parse_string,  0, &arg_dependency_style[UNIT_REQUIRES]  },
+                { "Style", "Requisite",         config_parse_string,  0, &arg_dependency_style[UNIT_REQUISITE] },
+                { "Style", "Wants",             config_parse_string,  0, &arg_dependency_style[UNIT_WANTS]     },
+                { "Style", "Conflicts",         config_parse_string,  0, &arg_dependency_style[UNIT_CONFLICTS] },
+                { "Style", "BindsTo",           config_parse_string,  0, &arg_dependency_style[UNIT_BINDS_TO]  },
+                { "Style", "PartOf",            config_parse_string,  0, &arg_dependency_style[UNIT_PART_OF]   },
+                { "Style", "OnFailure",         config_parse_string,  0, &arg_dependency_style[UNIT_ON_FAILURE]},
+                { "Style", "Triggers",          config_parse_string,  0, &arg_dependency_style[UNIT_TRIGGERS]  },
+                {}
+        };
+
+        static const char* colors[] = {
+                "black",    /* UNIT_REQUIRES */
+                "blue",     /* UNIT_REQUISITE */
+                "grey66",   /* UNIT_WANTS */
+                "magenta",  /* UNIT_BINDS_TO */
+                "black",    /* UNIT_PART_OF */
+                NULL,
+                NULL,
+                NULL,
+                NULL,
+                NULL,
+                "red",      /* UNIT_CONFLICTS */
+                NULL,
+                NULL,
+                "green",    /* UNIT_AFTER */
+                "orange",   /* UNIT_ON_FAILURE */
+                "grey66"    /* UNIT_TRIGGERS */
+        };
+
+        for (UnitDependency dependency = UNIT_REQUIRES; dependency <= UNIT_TRIGGERS; ++dependency) {
+                if (colors[dependency]) {
+                        arg_dependency_color[dependency] = strdup(colors[dependency]);
```

Comment:
this will override whatever has been configured in the configuration file/cmdline, which is not great I think

Reply:
No, these are the defaults, see comments above.

Reply:
changed